settings.gradle

```diff
@@ -1,15 +1,17 @@
 
 pluginManagement {
-    repositories {
-        maven {
-            url "${artifactory_contextUrl}/plugins-release"
-        }
-        if (gradlePluginsVersion.contains("SNAPSHOT") || versioningPluginVersion.contains("SNAPSHOT")) {
-            maven {
-                url "${artifactory_contextUrl}/plugins-snapshot-local"
-            }
-        }
-    }
+    // This doesn't seem necessary. Remove?
+//    repositories {
+//        maven {
+//            url "${artifactory_contextUrl}/plugins-release"
+//        }
+//        if (gradlePluginsVersion.contains("SNAPSHOT") || versioningPluginVersion.contains("SNAPSHOT")) {
+//            mavenLocal()
+//            maven {
+//                url "${artifactory_contextUrl}/plugins-snapshot-local"
+//            }
+//        }
+//    }
```

Comment:
I'm surprised if it's not necessary. My read of things is that Gradle will not look in our custom repository for plugins unless we configure them. Perhaps the declaration of the repositories in the `buildScript` block applies here as well, but that seems counterintuitive to me.

Reply:
It seems to fall back to the `buildscript` repositories if this is removed. Mangling the URLs in `pluginManagement.repositories` does produce an error, so it isn't ignored entirely.

I'm not sure the `pluginManagement.plugins` block is doing anything at all though. I can remove plugins, reference non-existent plugins, reference non-existent versions, or change plugins to `apply true`; all with no apparent negative impact on my build.
I think adding `gradlePlugins` to the classpath below overrides any version you might specify in `pluginManagement`.

Reply:
The `pluginManagement` block is definitely the more modern way to do things, but I'm unclear on the benefits, other than knowing it is the modern syntax. If we take out the classpath dependency and leave the `pluginManagement` block (including a use of `mavenLocal`), does it still work well, even with a locally published version of the plugins?

Reply:
If I remove the classpath dependency, it fails because it can't find `BuildUtils`.
It seems like custom types like BuildUtils or our various task types can only be brought in via classpath. The docs imply that you can pull in custom types via plugin but the only [example of importing a custom task type](https://docs.gradle.org/current/userguide/custom_tasks.html#sec:using_your_task_class_in_another_project) does so using classpath.
It looks like we might be able to [wrap `BuildUtils` in an extension](https://docs.gradle.org/current/userguide/custom_gradle_types.html#nested_objects) and expose that through a plugin but I don't think it's worth the trouble.
Having our plugin jar on the buildscript classpath seems to override everything referencing those plugins in the `pluginManagement` block.
I fiddled with it some more and found a scenario where the `pluginManagement` block matters. If we want to pull in the 'versioning' plugin without adding it to the classpath, pluginManagement has to have the correct repositories.